pipeline/processor/postprocessor.py

```diff
@@ -0,0 +1,268 @@
+    handled by the API itself)."""
+
+    regex_pattern = r"L\[?\d?\]?"
+
+    @staticmethod
+    def _minimal_bounding_box(coords: List[list]) -> list:
+        """
+        Return the minimally enclosing bounding box of bounding boxes.
+
+        Args:
+            coords: A list of coordinates for each bounding box formatted [x1,y1,x2,y2] with the top left as the origin.
+
+        Returns:
+            A list of coordinates for the minimally enclosing bounding box for all input bounding boxes.
+
+        """
+        x_min = min(coord[0] for coord in coords)
+        y_min = min(coord[1] for coord in coords)
+        x_max = max(coord[2] for coord in coords)
+        y_max = max(coord[3] for coord in coords)
+        return [x_min, y_min, x_max, y_max]
+
+    @staticmethod
+    def _find_first_occurrence(regex_pattern: str, list_of_strings):
+        """
+        Finds the first occurrence of a regular expression in a list of strings.
+
+        Args:
+            regex_pattern: Regular expression to search for in the list of strings.
+            list_of_strings: List of strings to search through.
+
+        Returns:
+            The first occurrence of the regular expression in the list of strings.
+
+        """
+        for string in list_of_strings:
+            if re.search(regex_pattern, string):
+                return string
+        return None
+
+    @staticmethod
+    def _pprint_list(df: pd.DataFrame) -> str:
+        """
+        Pretty print a text block list so that
+        it can be viewed in a programming language
+        or text editor.
+
+        Args:
+            df:
+
+        Returns:
+
+        """
+        string = ""
+        for ix, row in df.iterrows():
+            if row["type"] == "Lbl":
+                string += f"{row['text'][0]} "
+            elif row["type"] == "LBody":
+                string += f"{row['text'][0]}\n"
+            else:
+                string += f"{row['text'][0]}\n"
+        return string
+
+    @staticmethod
+    def _update_custom_attributes(text_block: dict, new_attribute: str) -> dict:
+        """
+        Helper method to update custom attributes with metadata to inform that a block
+        is contiguous with the previous element, which may be of a different type.
+
+        Args:
+            text_block: The text block that is contiguous from the previous page.
+            new_attribute: The new attribute to add to custom_attributes.
+
+        Returns:
+            The text block with the updated custom attributes.
+
+        """
+        new_custom_attributes = {new_attribute: True}
+        if text_block["custom_attributes"] is not None:
+            text_block["custom_attributes"].update(new_custom_attributes)
+        else:
+            text_block["custom_attributes"] = new_custom_attributes
+        return text_block
+
+    @staticmethod
+    def _postprocess_list_grouped_page(text_blocks: List[dict]) -> dict:
```

Comment:
Maybe this function would be better renamed to something along the lines of `_remove_unmerged_lists`, as it seems to do only that?